File: 0069-sqrtx/0069-sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x<2:
            return x
        left=1
        right=x//2
        while left<=right:
            mid=left+((right-left)//2)
            if mid*mid==x:
                return mid
            elif mid*mid > x:
                right=mid-1
            else:
                left=mid+1
        return right

File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        if(x==0)    return 0; 
        for(int i=1;i<=x/2+1;i++){
            if(i>INT_MAX/i)    return i-1;
            else if(i*i==x) return i;
            else if(i*i>x) return i-1;
        }
        return 0;
    }
};